internal/watcher/minio.go

```diff
@@ -91,15 +99,21 @@ func (w *minioWatcher) Path() string {
 	return ""
 }
 
-func (w *minioWatcher) blpop(ctx context.Context) (*BlobEvent, error) {
-	val, err := w.client.BLPop(ctx, redisPopTimeout, w.listName).Result()
+func (w *minioWatcher) pop(ctx context.Context) (*BlobEvent, error) {
+	val, err := w.client.BLMove(ctx, w.listName, w.failedList, "RIGHT", "LEFT", redisPopTimeout).Result()
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving from Redis list: %w", err)
 	}
 
-	event, err := w.event(val[1])
+	event, err := w.event(val)
 	if err != nil {
-		return nil, fmt.Errorf("ereror processing item received: %w", err)
+		return nil, fmt.Errorf("error processing item received: %w", err)
+	}
+
+	if c, err := w.client.LRem(ctx, w.failedList, 1, val).Result(); err != nil {
+		w.logger.Error(err, "Error removing message from failed list.", "list", w.failedList)
+	} else {
+		w.logger.V(2).Info("Successfully removed %d messages", "amount", c)
```

Comment:
I've been logging full sentences with the ending period and all, does that make sense? When I see the ending period missing then I can only think it's trying to be a heading. Not sure if the plural form is useful, it's always going to be `1` unless it errors.